apps/studio/pages/api/ai/sql/generate-v4.ts

```diff
@@ -79,7 +80,14 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse) {
     return res.status(400).json({ error: 'Invalid request body', issues: parseError.issues })
   }
 
-  const { messages: rawMessages, projectRef, connectionString, orgSlug, chatName } = data
+  const {
+    messages: rawMessages,
+    projectRef,
+    connectionString,
+    orgSlug,
+    chatName,
+    model: requestedModel,
```

Comment:
I don't know if we care about this, but even though the model selection is gated on plan, we have no checks to enforce it